LanguageFeatures/Augmentation-libraries/augmenting_declarations_A02_t01.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - An augmenting declaration appears in a library before the library where
+///   the original declaration occurs, according to merge order.
+///
+/// @description Checks that it is a compile-time error if an augmenting
+/// declaration appears before non-augmenting one. Test augmented declaration in
+/// a main library
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmenting_declarations_A02_t01_lib.dart';
+
+/**/augment class C {}
```

Comment:
If they accept #3647 then this would be a different error, in which case the description would need to be adjusted.

Reply:
> If they accept #3647 then this would be a different error, in which case the description would need to be adjusted.

#3647 was accepted. Why this is a different error now?

Reply:
The outcome of #3647 would make a difference for the error message (we'd have "error: augmenting declaration in library" if rejected and "error: augmenting declaration before corresponding original declaration" if accepted).

But I got it wrong: The description fits the 'accepted' case, which is the one that we got.